### DISCUSSION TITLE
Version 7.0.0

### DISCUSSION THREAD

**Discussion by Shane32** [graphql-dotnet/server#896] [General]
## What's New

* Entirely new implementation of WebSocket / subscription support
* Support for authorization over WebSocket connections
* Refactored implementation of ASP.NET Core middleware for better extensibility
* Support for the application/graphql+json media type
* Support for ASP.NET Core 2.1 running on .NET Framework 4.8 or .NET Core 2.1
* Entirely new authorization rule with support for `AllowAnonymous` and role-based rules
* Support for endpoint authorization rules
* Configuration simplified to a single line of code
* New readme with much better instructions and usage notes
* Many new samples added, such as usage with JWT Bearer tokens
* Reduced dependencies on third-party packages
* Many new options for configuration of GraphQL endpoint
* NRT annotations and rewritten XML comments throughout

See [Migrating from 6.x to 7.x](https://github.com/graphql-dotnet/server/blob/master/docs/migration/migration7.md) for detailed migration notes.

**Full Changelog**: https://github.com/graphql-dotnet/server/compare/6.1.1...7.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/graphql-dotnet/server/releases/tag/7.0.0'>Version 7.0.0</a>.</em>

**Comment by mariusmuntean** [comment]
I just upgraded from 6.x to 7.1 using the migration guide. The upgrade went smooth!
There's now less conf code and I could finally switch to graphql-transport-ws.

Many thanks!

**Reply by sungam3r** [reply]
Thanks for feedback.
